src/server/hset_family.cc

```diff
@@ -653,6 +662,11 @@ OpResult<uint32_t> OpSet(const OpArgs& op_args, string_view key, CmdArgList valu
 
   if (lp) {
     bool inserted;
+    size_t malloc_reserved = zmalloc_size(lp);
+    size_t min_sz = EstimateListpackMinBytes(values);
+    if (min_sz > malloc_reserved) {
+      lp = (uint8_t*)zrealloc(lp, min_sz);
```

Comment:
Can `zrealloc` fail here if it fails to expand the current memory block for that pointer?

Reply:
zrealloc like realloc does not guarantee pointer preservation. This is why it returns a pointer - could be a new one or the old one.